### DISCUSSION TITLE
Using front matter in non-template files

### DISCUSSION THREAD

**Discussion by Nixinova** [11ty/eleventy#1631] [Q&A]
I have some files in `/pages/_txt` ([repo](https://github.com/Nixinova/website/tree/b5b52a0c43170f9e5d28ba93aed87374f0453072)) that I want to rewrite to the root of my site. The only way I have found to do this is to set the extension to `.njk` (e.g.), but this messes with my language information.

I have tried adding e.g. `txt` to `setTemplateFormats` in `.eleventy.js` but that doesn't work.
Just having the front matter in the file by itself also doesn't work.

So how can I use front matter in non-template files?

**Comment by pdehaan** [comment]
What do you mean by “messes with my language information “?
11ty only supports specific template formats currently, so I don’t think setting it to “txt” would work, so I think your current solution of using a .njk (or .liquid, or .md) is probably best.

**Reply by Nixinova** [reply]
Messes with GitHub's language detection, the language configured in my editor, etc. Not 11ty.
